src/components/table-hoc/TableHOCUtils.ts

```diff
@@ -34,9 +40,17 @@ const getCompositeState = (id: string, state: IReactVaporState): ITableHOCCompos
 
         // filter
         filter: filter && filter.filterText,
+
+        // predicates
+        predicates: predicates,
     };
 };
 
+const getPredicateId = (tableId: string, componentId: string) => tableId + componentId;
+const getPaginationId = (tableId: string) => `pagination-${tableId}`;
+

```

Comment:
do you think its a good idea to add a util with something like "extractTablePredicate" and add the _.chain under in this utils ?